src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -247,6 +247,8 @@ public int DefaultBufferSize
             }
         }
 
+        internal bool IsInitializedForReflectionSerializer { get; set; }
```

Comment:
Perhaps add a comment documenting that the flag is intended to avoid always calling the `InitializeForReflectionSerializer` method on the hot path.